syntax = "proto3";

// option optimize_for = CODE_SIZE;
option cc_enable_arenas = true;

package bftrader.bfhisdata;

import "bftrader.proto";

// 历史数据服务
service BfHisDataService{

    // 保存tick，1分钟一打包，(todo:改成req，增加key需要的参数，ticks直接发bytes)
    rpc InsertTickPack(BfTickPackData) returns (BfVoid) {}
    
    // 保存bar (todo:改成req，增加key需要的参数，bar直接发bytes)
    rpc InsertBar(BfBarData) returns (BfVoid) {}
    
    // 获取tick
    rpc GetTickPack(BfGetTickPackReq) returns (stream BfTickPackData){}
    
    // 获取bar
    rpc GetBar(BfGetBarReq) returns (stream BfBarData) {}
}
